### ISSUE TITLE
Support for positional arguments?

### ISSUE THREAD

**Issue by nickeldan** [SupImDos/pydantic-argparse#17]
Is there currently a way to allow an argument to be passed positionally?  E.g.,

```python
class Args(BaseModel):
    x: str = pydantic.Field(description='The x.', positional=True)
    y: int = pydantic.Field(description='The y.')
```

```sh
./foo.py hello --y 5
```

**Comment by SupImDos** [comment]
Hi @nickeldan

Thanks for the question!

When I originally made `pydantic-argparse`, I decided against supporting positional arguments - mainly for simplicity in both interface design and handling of the arguments.

Specifically, see: https://pydantic-argparse.supimdos.com/background/#opinionated-design

The main issue I saw with positional arguments was:
1. Defining which argument types to allow them for.
    * This makes sense: `x: str = pydantic.Field(positional=True)`
    * This doesn't really: `z: bool = pydantic.Field(positional=True)`
    * i.e., they don't make sense for variadic arguments or flag arguments
2. How to handle the scenario where a user specifies a positional argument for an invalid type.
    * `pydantic-argparse` aims to be type-safe, and show issues *before* runtime
    * This issue would only be apparent at runtime: `z: bool = pydantic.Field(positional=True)`
    * i.e., not so type safe anymore - runtime errors, etc.

I may have overestimated with *how* complicated this may be, and I think with proper specifications this could be a useful feature.

**TLDR**: Not at the moment (see: [link](https://pydantic-argparse.supimdos.com/background/#opinionated-design)), maybe in the future? Work required on specification and implementation.

**Comment by nickeldan** [comment]
Makes sense.  Thanks!
